lib/ios-log.js

```diff
@@ -0,0 +1,208 @@
+        let limDir = path.resolve(__dirname, '../../../build/deviceconsole');
+        spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+        spawnEnv.DYLD_LIBRARY_PATH = `${limDir}:${process.env.DYLD_LIBRARY_PATH}`;
+        this.proc =  new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      }
+      this.finishStartingLogCapture();
+      return;
+    } 
+
+    let xcodeVersion = await xcode.getVersion();
+    let ver = parseInt(xcodeVersion.split('.')[0], 10);
+
+    if ( ver < 5 ) {
+      logger.debug('Starting iOS6.* simulator log capture');
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', '/var/log/system.log']);
+      this.finishStartingLogCapture();
+      return;
+    }
+
+    let logsPath;
+
+    if (ver >= 6) { 
+      logger.debug('Starting iOS 8.* simulator log capture');
+        if (_.isUndefined(this.simUdid)) {
+          logger.errorAndThrow('ios8 log capture requires a sim udid'); 
+        }
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+    } else {
+        logger.debug('Starting iOS7.* simulator log capture');
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+    }
+
+    try {
+      if (logsPath.indexOf('*') >= 0) {
+        return; 
+      }
+        await mkdirp(logsPath);
+        await touch(path.resolve(logsPath, 'system.log'));
+        await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+              'A new Simulator is about to start!'); 
+        let files;
+        try {
+          files = await glob(path.resolve(logsPath, 'system.log'));
+        if (files.length < 1) {
+          throw 'Could not start log capture';
+        }
+      } catch (e) {
+        logger.error('Could not start log capture because no iOS ' +
+        'simulator logs could be found at ' + logsPath + '/system.log' +
+        'Logging will not be functional for this run');
+      }
+
+      let lastModifiedLogPath = files[0];
+      let lastModifiedLogTime = await fs.stat(files[0]).mtime;
+      for (let file in files) {
+        let mtime = await fs.stat(file).mtime;
+        if (mtime > lastModifiedLogTime) {
+          lastModifiedLogPath = file;
+          lastModifiedLogTime = mtime;
+        }
+      }
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+      this.finishStartingLogCapture();
+    } catch (err) {
+      logger.warn('System log capture failed.');
+      return;
+    }    
+  }  
+
+  finishStartingLogCapture () {
+    if (!this.proc) {
+      let msg = 'Could not capture device log';
+      logger.warn(msg);
+      throw new Error(msg);
+    }
+     this.startTimer = setTimeout(function () {
+      let msg = 'Log capture did not start in a reasonable amount of time';
+      logger.error(msg);
+
+      throw new Error(msg);
+      
+    }, START_TIMEOUT);
+    let firstLine = true;
+    return new Promise((resolve, reject) => {
```

Comment:
seeing this line makes me think that you're not using the `SubProcess` API correctly; docs on it are here: https://github.com/appium/node-teen_process#teen_processsubprocess

let's touch base (with me or @imurchie) tomorrow to go over how it works in more detail